### PR TITLE
executor,distsql,util: distinguish walltime from sum of walltime in execution info

### LINKED ISSUE #56746: Systematically distinguish accumulated parallel time from walltime
## Enhancement
Currently, the "time" field in execution info may represent the execution time of a specific executor or the total execution time of multiple "cloned" executors which are executed parallely, such as ParallelApply cases：
image 

This situation is very confusing, since the Limit_17 operator's time is much more than its parent operator's time. So it is necessary to distinguish between these two meanings. At the same time, it is best to have a systematic and unified method to avoid processing each operator independently.

### PR DESCRIPTION
**Pull request by yibin87**
<!--

Thank you for contributing to TiDB!

PR Title Format:
1. pkg [, pkg2, pkg3]: what's changed
2. *: what's changed

-->

### What problem does this PR solve?
<!--

Please create an issue first to describe the problem.

There MUST be one line starting with "Issue Number:  " and
linking the relevant issues via the "close" or "ref".

For more info, check https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/contribute-code.html#referring-to-an-issue.

-->

Issue Number: close #56746 

Problem Summary:

### What changed and how does it work?
Thanks to Keao's [PR ](https://github.com/pingcap/tidb/pull/56485), all executors must be created by either NewBaseExecutorV2 or BuildNewBaseExecutorV2. Thus, we can identify the "parallel time" case by checking if any existing executors have the same ID. Then use a different name, like "total_parallel_time" to display it.
Some special cases:
1. IndexLookUp & IndexMergeReader 's index plan doesn't mapping to an Executor, just mapping to several distsql calls. So need to handle them case by case.
2. PointGet's Init function not only initializes some fields but also re-constructs its BaseExecutor, which is not reasonable, split it to two functions now.

**ParallelApply & IndexLookUp Query**
```
+--------------------------------------------+---------+---------+-----------+------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------+-----------+------+
| id                                         | estRows | actRows | task      | access object          | execution info                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         | operator info                                                                  | memory    | disk |
+--------------------------------------------+---------+---------+-----------+------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------+-----------+------+
| Projection_11                              | 10.00   | 2       | root      |                        | time:2.77ms, loops:3, RU:2.480302, Concurrency:OFF                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     | Column#13                                                                      | 2.14 KB   | N/A  |
| └─Apply_13                                 | 10.00   | 2       | root      |                        | time:2.76ms, loops:3, Concurrency:5, cache:ON, cacheHitRatio:0.000%                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    | CARTESIAN left outer join                                                      | 778 Bytes | N/A  |
|   ├─TableReader_15(Build)                  | 10.00   | 2       | root      |                        | time:656.7µs, loops:2, cop_task: {num: 1, max: 557.5µs, proc_keys: 2, tot_proc: 64.4µs, tot_wait: 60.7µs, copr_cache_hit_ratio: 0.00, build_task_duration: 6.45µs, max_distsql_concurrency: 1}, rpc_info:{Cop:{num_rpc:1, total_time:537.1µs}}                                                                                                                                                                                                                                                                                                                                                                                                                                         | data:TableRangeScan_14                                                         | 286 Bytes | N/A  |
|   │ └─TableRangeScan_14                    | 10.00   | 2       | cop[tikv] | table:alias            | tikv_task:{time:0s, loops:1}, scan_detail: {total_process_keys: 2, total_process_keys_size: 100, total_keys: 3, get_snapshot_time: 40µs, rocksdb: {key_skipped_count: 2, block: {}}}, time_detail: {total_process_time: 64.4µs, total_wait_time: 60.7µs, tikv_wall_time: 244.6µs}                                                                                                                                                                                                                                                                                                                                                                                                      | range:[1,1], keep order:false, stats:pseudo                                    | N/A       | N/A  |
|   └─MaxOneRow_16(Probe)                    | 10.00   | 2       | root      |                        | total_time:3.93ms, loops:4                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             |                                                                                | N/A       | N/A  |
|     └─StreamAgg_21                         | 10.00   | 2       | root      |                        | total_time:3.91ms, loops:4                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             | funcs:sum(Column#17)->Column#13                                                | 1.45 KB   | N/A  |
|       └─Projection_48                      | 100.00  | 2       | root      |                        | total_time:3.87ms, loops:4, Concurrency:OFF                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            | cast(test.t1.c4, decimal(10,0) BINARY)->Column#17                              | 760 Bytes | N/A  |
|         └─Projection_47                    | 100.00  | 2       | root      |                        | total_time:3.81ms, loops:4, Concurrency:OFF                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            | test.t1.c3, test.t1.c4                                                         | 1.48 KB   | N/A  |
|           └─IndexLookUp_46                 | 100.00  | 2       | root      |                        | total_time:3.79ms, loops:4, index_task: {total_time: 1.68ms, fetch_handle: 1.67ms, build: 2.14µs, wait: 7.05µs}, table_task: {total_time: 1.83ms, num: 2, concurrency: 5}, next: {wait_index: 1.84ms, wait_table_lookup_build: 271.8µs, wait_table_lookup_resp: 1.55ms}                                                                                                                                                                                                                                                                                                                                                                                                                |                                                                                | 18.5 KB   | N/A  |
|             ├─IndexRangeScan_44(Build)     | 100.00  | 2       | cop[tikv] | table:t1, index:c3(c3) | total_time:1.64ms, loops:6, cop_task: {num: 2, max: 750.8µs, min: 727µs, avg: 738.9µs, p95: 750.8µs, max_proc_keys: 1, p95_proc_keys: 1, tot_proc: 118µs, tot_wait: 188.2µs, copr_cache_hit_ratio: 0.00, build_task_duration: 25.3µs, max_distsql_concurrency: 1}, rpc_info:{Cop:{num_rpc:2, total_time:1.45ms}}, tikv_task:{proc max:0s, min:0s, avg: 0s, p80:0s, p95:0s, iters:2, tasks:2}, scan_detail: {total_process_keys: 2, total_process_keys_size: 114, total_keys: 4, get_snapshot_time: 120.4µs, rocksdb: {key_skipped_count: 2, block: {}}}, time_detail: {total_process_time: 118µs, total_wait_time: 188.2µs, tikv_wall_time: 743.4µs}                                   | range: decided by [eq(test.t1.c3, test.t1.c3)], keep order:false, stats:pseudo | N/A       | N/A  |
|             └─TableRowIDScan_45(Probe)     | 100.00  | 2       | cop[tikv] | table:t1               | total_time:1.51ms, loops:4, cop_task: {num: 2, max: 665µs, min: 649.1µs, avg: 657µs, p95: 665µs, max_proc_keys: 1, p95_proc_keys: 1, tot_proc: 119.2µs, tot_wait: 170µs, copr_cache_hit_ratio: 0.00, build_task_duration: 149.7µs, max_distsql_concurrency: 1, max_extra_concurrency: 1}, rpc_info:{Cop:{num_rpc:2, total_time:1.28ms}}, tikv_task:{proc max:0s, min:0s, avg: 0s, p80:0s, p95:0s, iters:2, tasks:2}, scan_detail: {total_process_keys: 2, total_process_keys_size: 100, total_keys: 4, get_snapshot_time: 114.6µs, rocksdb: {key_skipped_count: 2, block: {}}}, time_detail: {total_process_time: 119.2µs, total_wait_time: 170µs, tikv_wall_time: 629.1µs}            | keep order:false, stats:pseudo                                                 | N/A       | N/A  |
+--------------------------------------------+---------+---------+-----------+------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------+-----------+------+
```

**IndexMergeReader Query**
```
+-------------------------------+----------+---------+-----------+------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+---------------------------------------------------------------------+---------+------+
| id                            | estRows  | actRows | task      | access object          | execution info                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   | operator info                                                       | memory  | disk |
+-------------------------------+----------+---------+-----------+------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+---------------------------------------------------------------------+---------+------+
| IndexMerge_8                  | 18959.99 | 18959   | root      |                        | time:111.1ms, loops:20, RU:303.684901, index_task:{fetch_handle:13.903218ms, merge:4.080369ms}, table_task:{num:2, concurrency:5, fetch_row:199.93441ms, wait_time:20.396937ms}                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  | type: union                                                         | 1.14 MB | N/A  |
| ├─IndexRangeScan_5(Build)     | 9997.45  | 9978    | cop[tikv] | table:t, index:idx1(b) | time:6.38ms, loops:12, cop_task: {num: 1, max: 6.19ms, proc_keys: 9978, tot_proc: 5.19ms, tot_wait: 65.3µs, copr_cache_hit_ratio: 0.00, build_task_duration: 23.5µs, max_distsql_concurrency: 1}, rpc_info:{Cop:{num_rpc:1, total_time:6.15ms}}, tikv_task:{time:4ms, loops:14}, scan_detail: {total_process_keys: 9978, total_process_keys_size: 458988, total_keys: 9979, get_snapshot_time: 45.1µs, rocksdb: {key_skipped_count: 9978, block: {cache_hit_count: 2}}}, time_detail: {total_process_time: 5.19ms, total_suspend_time: 16.5µs, total_wait_time: 65.3µs, total_kv_read_wall_time: 4ms, tikv_wall_time: 5.49ms}                                                                                                                                                                                                                                    | range:[-inf,100000), keep order:false, stats:partial[id:allEvicted] | N/A     | N/A  |
| ├─IndexRangeScan_6(Build)     | 9958.09  | 9947    | cop[tikv] | table:t, index:idx2(c) | time:6.4ms, loops:12, cop_task: {num: 1, max: 6.23ms, proc_keys: 9947, tot_proc: 5.48ms, tot_wait: 37.5µs, copr_cache_hit_ratio: 0.00, build_task_duration: 23.5µs, max_distsql_concurrency: 1}, rpc_info:{Cop:{num_rpc:1, total_time:6.21ms}}, tikv_task:{time:5ms, loops:14}, scan_detail: {total_process_keys: 9947, total_process_keys_size: 457562, total_keys: 9948, get_snapshot_time: 25.6µs, rocksdb: {key_skipped_count: 9947, block: {cache_hit_count: 2}}}, time_detail: {total_process_time: 5.48ms, total_suspend_time: 18.8µs, total_wait_time: 37.5µs, total_kv_read_wall_time: 5ms, tikv_wall_time: 5.69ms}                                                                                                                                                                                                                                     | range:[-inf,100000), keep order:false, stats:partial[id:allEvicted] | N/A     | N/A  |
| └─TableRowIDScan_7(Probe)     | 18959.99 | 18959   | cop[tikv] | table:t                | total_time:188.9ms, loops:21, cop_task: {num: 200, max: 19.7ms, min: 8.96ms, avg: 13.7ms, p95: 18.4ms, max_proc_keys: 122, p95_proc_keys: 112, tot_proc: 530.4ms, tot_wait: 648.3ms, copr_cache_hit_ratio: 0.00, build_task_duration: 1.68ms, max_distsql_concurrency: 15}, rpc_info:{Cop:{num_rpc:200, total_time:2.73s}}, tikv_task:{proc max:16ms, min:4ms, avg: 9.73ms, p80:10ms, p95:15ms, iters:494, tasks:200}, scan_detail: {total_process_keys: 18959, total_process_keys_size: 877859, total_keys: 26786, get_snapshot_time: 6.43ms, rocksdb: {delete_skipped_count: 1434, key_skipped_count: 11028, block: {cache_hit_count: 11416, read_count: 273, read_byte: 4.24 MB, read_time: 63.8ms}}}, time_detail: {total_process_time: 530.4ms, total_suspend_time: 1.42s, total_wait_time: 648.3ms, total_kv_read_wall_time: 1.95s, tikv_wall_time: 2.65s} | keep order:false, stats:partial[id:allEvicted]                      | N/A     | N/A  |
+-------------------------------+----------+---------+-----------+------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+---------------------------------------------------------------------+---------+------+
```
### Check List

Tests <!-- At least one of them must be included. -->

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > <!-- Or your custom  "No need to test" reasons -->

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

<!-- compatibility change, improvement, bugfix, and new feature need a release note -->

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by windtalker — APPROVED** [review]
lgtm

File: pkg/util/execdetails/execdetails.go [hunk 1]
```diff
@@ -1494,17 +1501,33 @@ func (e *RuntimeStatsColl) RegisterStats(planID int, info RuntimeStats) {
 	}
 }
 
-// GetBasicRuntimeStats gets basicRuntimeStats for a executor.
-func (e *RuntimeStatsColl) GetBasicRuntimeStats(planID int) *BasicRuntimeStats {
+// GetBasicRuntimeStats gets basicRuntimeStats for a executor
+// When rootStat/rootStat's basicRuntimeStats is nil, the behavior is decided by initNewExecutorStats argument:
+// 1. If true, it created a new one, and increase basicRuntimeStats' executorCount
+// 2. Else, it returns nil
+func (e *RuntimeStatsColl) GetBasicRuntimeStats(planID int, initNewExecutorStats bool) *BasicRuntimeStats {
```
Comment (XuHuaiyu) [inline pkg/util/execdetails/execdetails.go:1508]: Is the following code be more readable?

``` go
func (e *RuntimeStatsColl) GetBasicRuntimeStats(planID int, initNewExecutorStats bool) *BasicRuntimeStats {
	e.mu.Lock()
	defer e.mu.Unlock()

	stats, ok := e.rootStats[planID]
	if !ok && initNewExecutorStats {
		stats = NewRootRuntimeStats()
		e.rootStats[planID] = stats
	}

	if stats == nil {
		return nil
	}

	if stats.basic == nil && initNewExecutorStats {
		stats.basic = &BasicRuntimeStats{}
		stats.basic.executorCount.Add(1)
	} else if stats.basic != nil && initNewExecutorStats {
		stats.basic.executorCount.Add(1)
	}

	return stats.basic
}
```

Comment (yibin87) [reply pkg/util/execdetails/execdetails.go:1508]: I'll change it

**Review by XuHuaiyu — APPROVED** [review]

**Pull request merged**